pypeit/core/skysub.py

```diff
@@ -1210,23 +1210,29 @@ def ech_local_skysub_extract(sciimg, sciivar, fullmask, tilts, waveimg,
 
     # Find the spat IDs
     gdslit_spat = np.unique(slitmask[slitmask >= 0]).astype(int)  # Unique sorts
-    if gdslit_spat.size != norders:
-        msgs.error("You have not dealt with masked orders properly")
+    #if gdslit_spat.size != norders:
+    #    msgs.error("You have not dealt with masked orders properly")
 
-    if (np.sum(sobjs.sign > 0) % norders) == 0:
-        nobjs = int((np.sum(sobjs.sign > 0)/norders))
-    else:
-        msgs.error('Number of specobjs in sobjs is not an integer multiple of the number or ordres!')
+    #if (np.sum(sobjs.sign > 0) % norders) == 0:
+    #    nobjs = int((np.sum(sobjs.sign > 0)/norders))
+    #else:
+    #    msgs.error('Number of specobjs in sobjs is not an integer multiple of the number or ordres!')
 
-    order_snr = np.zeros((norders, nobjs))
+    # Set bad obj to -nan
     uni_objid = np.unique(sobjs[sobjs.sign > 0].ECH_OBJID)
+    nobjs = len(uni_objid)
+    order_snr = np.zeros((norders, nobjs))
     for iord in range(norders):
         for iobj in range(nobjs):
             ind = (sobjs.ECH_ORDERINDX == iord) & (sobjs.ECH_OBJID == uni_objid[iobj])
-            order_snr[iord,iobj] = sobjs[ind].ech_snr
+            # Allow for missed/bad order
```

Comment:
I agree with @kbwestfall here. Let's use a boolean rather than nan and nanmean

Reply:
done